### PR TITLE
feat(systemtags): add bulk tagging action

### LINKED ISSUE #45935: Add  Bulk Tagging on new Files app [28+]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
Hello everyone,

Since my recent update to Nextcloud 28, I’ve noticed a change in the file tagging functionality. Previously, I had the ability to tag multiple files simultaneously, but this feature seems to have disappeared since the update. This limitation is causing an inconvenience as I frequently use the tagging feature to organize my files.

**Describe the solution you'd like**
I would like to have the ability to tag multiple files simultaneously reinstated in Nextcloud 28.

**Describe alternatives you've considered**
Currently, the only alternative is to tag files individually, which is time-consuming and less efficient.

**Additional context**
Has anyone else encountered this issue? Is it a new feature introduced in version 28, or is it potentially a bug? If anyone has information on this matter or has found a solution, I would appreciate it if you could share. Otherwise, can we consider reporting this as a potential issue to the development team?

Thank you in advance for your responses and assistance!

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/45935

image

### CODE REVIEW

File: core/Migrations/Version31000Date20241018063111.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			if (!$table->hasIndex('systag_objecttype')) {
+				$table->addIndex(['objecttype'], 'systag_objecttype');
+			}
+		}
+
+		if ($schema->hasTable('systemtag')) {
+			$table = $schema->getTable('systemtag');
+
+			if (!$table->hasColumn('etag')) {
+				$table->addColumn('etag', 'string', [
```
Comment (provokateurin) [inline core/Migrations/Version31000Date20241018063111.php:44]:
```suggestion
				$table->addColumn('etag', Types::STRING, [
```

**Review by provokateurin — APPROVED** [review]
PHP looks good to me, just a minor suggestion

**Review by susnux — APPROVED** [review]

**Comment by rolandinus** [comment]
I notice that when Tags are assigned or unassigned via this bulk tagging action, the Events MapperEvent::EVENT_ASSIGN, and MapperEvent::EVENT_UNASSIGN are not dispatched. This might cause problems with any apps which rely on it. 
Ist there another way we can track changed tags?

**Pull request merged**